runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/HotKeyLoggerTest.java

```diff
@@ -50,18 +45,29 @@ public void SetUp() {
   }
 
   @Test
-  public void correctHotKeyMessage() {
+  public void correctHotKeyMessageWithoutKey() {
     HotKeyLogger hotKeyLogger = new HotKeyLogger(clock);
 
-    assertFalse(hotKeyLogger.isThrottled());
-    String m = hotKeyLogger.getHotKeyMessage("TEST_STEP_ID", "1s");
+    hotKeyLogger.logHotKeyDetection("TEST_STEP_ID", Duration.standardSeconds(1));
     assertTrue(hotKeyLogger.isThrottled());
 
-    assertEquals(
+    expectedLogs.verifyWarn(
         "A hot key was detected in step 'TEST_STEP_ID' with age of '1s'. This is a "
             + "symptom of key distribution being skewed. To fix, please inspect your data and "
-            + "pipeline to ensure that elements are evenly distributed across your key space.",
-        m);
+            + "pipeline to ensure that elements are evenly distributed across your key space.");
+  }
+
+  @Test
+  public void correctHotKeyMessageWithKey() {
```

Comment:
might as well cover the `null` as an object case to ensure we don't regress.

Reply:
Done